homeassistant/components/qnap/sensor.py

```diff
@@ -208,34 +201,29 @@ def setup_platform(
     discovery_info: DiscoveryInfoType | None = None,
 ) -> None:
     """Set up the QNAP NAS sensor."""
-    api = QNAPStatsAPI(config)
-    api.update()
-
-    # QNAP is not available
-    if not api.data:
-        raise PlatformNotReady
+    coordinator = hass.data[DOMAIN][config.entry_id]
```

Comment:
Note: this still needs to be adjusted.
You'll have to initialise the coordinator here, try run an initial update, and rasie PlatformNotReady if it failed.